GameCenterAuthProvider+Combine.swift

```diff
@@ -0,0 +1,43 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#if canImport(Combine) && swift(>=5.0) && canImport(FirebaseAuth)
+
+  import Combine
+  import FirebaseAuth
+
+  @available(swift 5.0)
+  @available(macOS 10.15, iOS 13, watchOS 6, tvOS 13, *)
```

Comment:
According to https://github.com/firebase/firebase-ios-sdk/blob/master/FirebaseAuth/Sources/Public/FirebaseAuth/FIRGameCenterAuthProvider.h#L47, GameCenterAuthProvider is not available on watchOS.

@rosalyntan according to [Apple's docs](https://developer.apple.com/documentation/gamekit/gklocalplayer), GKPlayer is available on watchOS. Can you comment on whether there are other reasons why we specifically exclude watchOS in [this line](https://github.com/firebase/firebase-ios-sdk/blob/master/FirebaseAuth/Sources/Public/FirebaseAuth/FIRGameCenterAuthProvider.h#L47)?

Reply:
I believe watchOS is excluded because Apple deprecated the `playerID` property in watchOS 6, but the replacement `teamPlayerID` is not supported on watchOS.

Adding @ryanwilson who reviewed the PR that added watchOS support to auth.

Reply:
Good find, I believe that's the reason!

Reply:
Ok, so should we

1. completely remove watchOS from the list of supported OSes, 
2. or state that is supported up to watchOS 5?

Reply:
Remove. There's no reason to add support only for old versions.

Reply:
@ziadtamim, can you address this feedback, please?

Reply:
Done under https://github.com/firebase/firebase-ios-sdk/pull/7599